### PR TITLE
Add Base, Ref, and SHA to EditChange struct

### PR DESCRIPTION
**Pull request by JacobValdemar**
Add support for webhook which is triggered when you change the base branch in a pull request after opening it. 

Fixes: #1617

### CODE REVIEW

File: github/event_types.go [hunk 1]
```diff
@@ -221,6 +221,14 @@ type EditChange struct {
 	Body *struct {
 		From *string `json:"from,omitempty"`
 	} `json:"body,omitempty"`
+	Base *struct {
+		Ref *struct {
+			From *string `json:"from,omitempty"`
+		} `json:"ref,omitempty"`
+		Sha *struct {
```
Comment (gmlewis) [inline github/event_types.go:228]:
```suggestion
		SHA *struct {
```

for idiomatic Go.

Comment (JacobValdemar) [reply github/event_types.go]: https://github.com/google/go-github/pull/1619/commits/05744972613659beb36b076a03cf38f8446682fb

**Comment by JacobValdemar** [comment]
Good idea about the unit test @gmlewis. I can't locate where the other content in event_types.go are tested. Do you have any immediate position on where this test should be placed? 🙂

**Comment by gmlewis** [comment]
> I can't locate where the other content in event_types.go are tested. Do you have any immediate position on where this test should be placed?

Oh, wow!  You are right... I thought we had that.  

How about creating a new file, `event_types_test.go` with the appropriate header, and then take a look at #55 to see a large number of examples of PRs dedicated to just testing the marshaling and unmarshaling of the structs?  That would be fantastic.

Thank you, @JacobValdemar !

**Comment by JacobValdemar** [comment]
Hi again @gmlewis :) I've now added the unit tests you requested.

**Review by wesleimp — APPROVED** [review]
Thank you, @JacobValdemar 
LGTM 👌🏼

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Comment by jhlegarreta** [comment]
I'm wondering whether there is an estimated time of arrival of the feature in this PR to be shipped into a release. If I'm not wrong release `v33.0.0` did not include it.

Some downstream repositories (cross-referencing https://github.com/srvaroa/labeler/issues/13) would benefit from having it on a release. Thanks.

**Comment by gmlewis** [comment]
Hi @jhlegarreta - as of #1280, all releases (major, minor, and patch) are now on-demand.  (I should probably put this in the README.md... sorry.)

I won't be able to start the process quite yet, but I believe we could cut a new release later today (or early tomorrow) if I can get some assistance with code reviews.

**Comment by jhlegarreta** [comment]
@gmlewis thanks for the prompt answer. Wow, it's nice that you are able to cut releases on demand. I appreciate the effort.

**Comment by gmlewis** [comment]
Hi @jhlegarreta - https://github.com/google/go-github/releases/tag/v34.0.0 has been released.

**Comment by jhlegarreta** [comment]
Much appreciated @gmlewis ! Thanks.

If I am allowed to post here a question tangentially related to this PR (let me know if I should ask it elsewhere), I'd like to know whether the package supports `pull_request_target` webhook events. This PR was cross-referenced from https://github.com/srvaroa/labeler/issues/13 concerning that aspect, but having read the current code (e.g. see the webhook event [files here](https://github.com/google/go-github/tree/master/github), the [message mapping](https://github.com/google/go-github/blob/master/github/messages.go#L79) and the PR [series of structs](https://github.com/google/go-github/blob/master/github/event_types.go#L612)) I have not found support for the webhook.

Am I right or is the webhook supported using some other mechanism or path that I do not see?

Would it be possible to add it as time permits? A few prominent open source projects from Apache or the ITK ecosystem could possibly benefit from it.

Thanks.

**Comment by gmlewis** [comment]
Ah, yes, it looks like our repo doesn't support it yet.  I'll open an issue and invite volunteers to implement it.  Thanks, @jhlegarreta .

**Comment by gmlewis** [comment]
There is now a new issue to support it: #1834.

**Pull request merged**